src/Game/AttackButton.js

```diff
@@ -0,0 +1,118 @@
+import { useEffect, useState, useMemo } from "react";
+import { supabase } from "../supabase_client";
+import { generateSixSidedDice, generateFourSidedDice } from "../utils";
+
+const AttackButton = (props) => {
+  const [currentTurnPlayerIndex, setCurrentTurnPlayerIndex] = useState(0);
+  const {
+    totalPlayerData,
+    roomCode,
+    attackTarget,
+    playerName,
+    isPlayerEliminated,
+    setTotalPlayerData,
+  } = props;
+
+  useEffect(() => {
+    const fetchHitpoints = async () => {
+      const { data } = await supabase
+        .from("Players")
+        .select("name, Hitpoints")
+        .order("id", { ascending: true })
+        .eq("roomID", roomCode);
+      const playerHp = data.map((playerData) => {
+        return { name: playerData.name, Hitpoints: playerData.Hitpoints };
+      });
+      setTotalPlayerData(playerHp);
+    };
+    fetchHitpoints();
+  }, []);
+
+  useEffect(() => {
+    let mySubscription = supabase
+      .from("Players")
+      .on("UPDATE", (payload) => {
+        setTotalPlayerData((currentState) =>
+          updateHitpointValues(currentState, payload)
+        );
+      })
+      .subscribe();
+    return () => {
+      supabase.removeSubscription(mySubscription);
+    };
+  }, []);
```

Comment:
I think it might make more sense to leave these in the game page since thats where the state variables you're updating live.